### ISSUE TITLE
[FEAT] - Split news sources into regions

### ISSUE THREAD

**Issue by MizouziE** [Energy-Price-News-API/energy-prices-api#100]
### What feature would you like to see?

The list of sources has grown very nicely lately, so it might be handy to have another method of splitting them.

The same way we have endpoints for each individual news source, I would like to be able to separate them by region also. 

For example:
- Europe
- Asia
- Africa
- Americas

### Possible implementation

- We will need a new dynamic route made in the `src/routes/api/news.js` file.
- It will need to use the same middleware as the singleSource route.
- We will need a new controller in the `src/controller/api/newsAPIController.js` file, similar to singleSource also.
- The sources will each need a way to identify the region, possibly by adding another key `"region": "e.g. europe/asia..."` to the array for each.

### Additional information

We could even go a step further and do `/east` and `/west` endpoints that group two regions together, but I think the continental/region split is sufficient for now.

**Comment by MuminAhmadKhan** [comment]
hey @MizouziE I want to work on this.

**Comment by MizouziE** [comment]
@MuminAhmadKhan Sure man! The stuff you did for the sources-list was great and I think this should be similar. 

One thing I'd suggest (if you would like to) is to allocate the adding of regions to the sources objects to a new/separate issue as it will be very repetitive but should be simple, so a perfect opportunity for a new-comer. Perhaps do 2 or 3 just to test the controllers & routes you create and leave the rest to be updated after. 

Up to you though, if you'd rather do it yourself.

One other thing. Be sure to 'git pull' from the master branch and run npm install or update before starting as I made a few changes to package-lock.json yesterday.

**Comment by MuminAhmadKhan** [comment]
Should I add the regions in  a seperate json file  or  write them in the concerned file only ,   for checking whether the input is valid and displaying the relevant sources from a particular region?

**Comment by MizouziE** [comment]
@MuminAhmadKhan That's a very good point! If we have a separate "data/regions.json" then we can not only validate the region provided for a source, but also have another endpoint showing what regions we cover. 💡💡💡

Feel free to tackle this as part of this ticket or make a separate issue. I need to update ur contribution to include "ideas" cos this is great!

**Comment by MuminAhmadKhan** [comment]
I am thinking of paths /api/new/regions showing the regions covered and say /api/news/regions/europe showing sources covered in europe? What's your say?

**Comment by MizouziE** [comment]
@MuminAhmadKhan I agree that sounds like the most sensible approach. Nice structure. 👌

**Comment by MuminAhmadKhan** [comment]
Is the Issue done or is something pending?

**Comment by MizouziE** [comment]
> Is the Issue done or is something pending?

Done, I just forgot to link this to the PR to automatically close it.
